### PR TITLE
Initial commit:adding decompression ratio support

### PR DESCRIPTION
**Pull request by mrsaldana**

### CODE REVIEW

**Comment by mrsaldana** [comment]
#build

**Comment by mrsaldana** [comment]
#build

**Comment by mrsaldana** [comment]
#build

File: dev/com.ibm.ws.transport.http/resources/OSGI-INF/l10n/metatype.properties [hunk 1]
```diff
@@ -206,6 +206,12 @@ http.proxyredirect.httpsPort.desc=The (secure) port to redirect to. Incoming HTT
 http.autoDecompression=Auto decompression
 http.autoDecompression.desc=Specifies whether the HTTP Channel automatically decompresses incoming request body data.
 
+http.decompressionRatioLimit=Decompression ratio limit
+http.decompressionRatioLimit.desc=Specifies the maximum ratio of decompressed to compressed request body payload. This ratio is verified as the body is read by the HTTP channel, and if the decompressionTolerance is reached, the request is ended.
```
Comment (ManasiGandhi) [inline dev/com.ibm.ws.transport.http/resources/OSGI-INF/l10n/metatype.properties]: Change,
"Specifies the maximum ratio of decompressed to compressed request body payload. This ratio is verified as the body is read by the HTTP channel, and if the decompressionTolerance is reached, the request is ended."
to,
"Specifies the maximum ratio of decompressed to compressed request body payload. The HTTP channel reads the request body and verifies the ratio. The request ends if the decompressionTolerance is reached."

File: dev/com.ibm.ws.transport.http/resources/OSGI-INF/l10n/metatype.properties [hunk 2]
```diff
@@ -206,6 +206,12 @@ http.proxyredirect.httpsPort.desc=The (secure) port to redirect to. Incoming HTT
...
+
+http.decompressionTolerance=Decompression tolerance
+http.decompressionTolerance.desc=Specifies the maximum number of times the HTTP channel can hit the decompressionRatioLimit as the request body is decompressed before the request is ended.
```
Comment (ManasiGandhi) [inline dev/com.ibm.ws.transport.http/resources/OSGI-INF/l10n/metatype.properties]: Change,
"Specifies the maximum number of times the HTTP channel can hit the decompressionRatioLimit as the request body is decompressed before the request is ended."
to
"Specifies the maximum number of times the HTTP channel can reach the decompressionRatioLimit during request body decompression before the request ends."

File: dev/com.ibm.ws.transport.http/resources/OSGI-INF/l10n/metatype.properties (hunk 1 above)
Comment (anupag) [reply dev/com.ibm.ws.transport.http/resources/OSGI-INF/l10n/metatype.properties]: I think we should define the decompressionTolerance, 

The request ends if the maximum number of times the HTTP channel reach this ratio sepcified by decompressionTolerance

File: dev/com.ibm.ws.transport.http/resources/OSGI-INF/l10n/metatype.properties (hunk 2 above)
Comment (anupag) [reply dev/com.ibm.ws.transport.http/resources/OSGI-INF/l10n/metatype.properties]: same here explain decompressionRatioLimit

File: dev/com.ibm.ws.transport.http/resources/com/ibm/ws/http/channel/internal/resources/httpchannelmessages.nlsprops [hunk 3]
```diff
@@ -160,4 +160,8 @@ compression.duplicateOverwriteAndAdd.useraction=If the default configuration mus
 
 compression.unsupportedAlgorithm=CWWKT0033W: An attempt was made to configure {0} as the server-preferred algorithm. The supported values are gzip, x-gzip, zlib, deflate, identity, and none. The default configuration {1} is used instead.
 compression.unsupportedAlgorithm.explanation=The configuration is attempting to set an unsupported algorithm as the server-preferred algorithm to use for compression. 
-compression.unsupportedAlgorithm.useraction= When defining the value, ensure that it is set to gzip, x-gzip, deflate, zlib, identity, or none.
\ No newline at end of file
+compression.unsupportedAlgorithm.useraction= When defining the value, ensure that it is set to gzip, x-gzip, deflate, zlib, identity, or none.
+
+decompression.tolerance.reached=CWWKT0034E: Attempted to decompress a body with a ratio larger than tolerated.
```
Comment (ManasiGandhi) [inline dev/com.ibm.ws.transport.http/resources/com/ibm/ws/http/channel/internal/resources/httpchannelmessages.nlsprops]: Change,
"Attempted to decompress a body with a ratio larger than tolerated."
to,
"The HTTP channel attempted to decompress a body with a larger than tolerated ratio."

File: dev/com.ibm.ws.transport.http/resources/com/ibm/ws/http/channel/internal/resources/httpchannelmessages.nlsprops [hunk 4]
```diff
@@ -160,4 +160,8 @@ compression.duplicateOverwriteAndAdd.useraction=If the default configuration mus
...
+decompression.tolerance.reached.explanation=The amount of tolerated decompression cycles, defined by the 'decompressionTolerance' httpOption attribute, which resulted in a decompression ratio larger than the configured limit, has been reached. The decompression limit is defined by the 'decompressionRatioLimit' httpOption attribute.
```
Comment (ManasiGandhi) [inline dev/com.ibm.ws.transport.http/resources/com/ibm/ws/http/channel/internal/resources/httpchannelmessages.nlsprops]: Change,
"The amount of tolerated decompression cycles, defined by the 'decompressionTolerance' httpOption attribute, which resulted in a decompression ratio larger than the configured limit, has been reached. The decompression limit is defined by the 'decompressionRatioLimit' httpOption attribute."
to,
"The maximum number of decompression cycles has reached, resulting in a decompression ratio larger than the configured limit. The decompression limit is defined by the 'decompressionRatioLimit' httpOption attribute."

File: dev/com.ibm.ws.transport.http/resources/com/ibm/ws/http/channel/internal/resources/httpchannelmessages.nlsprops [hunk 5]
```diff
@@ -160,4 +160,8 @@ compression.duplicateOverwriteAndAdd.useraction=If the default configuration mus
...
+decompression.tolerance.reached.useraction=When this exception happens on a request body that is considered valid, consider reconfiguring the 'decompressionTolerance' and 'decompressionRatioLimit' httpOption attributes to achieve the desired decompression results.
```
Comment (ManasiGandhi) [inline dev/com.ibm.ws.transport.http/resources/com/ibm/ws/http/channel/internal/resources/httpchannelmessages.nlsprops]: Change,
"When this exception happens on a request body that is considered valid, consider reconfiguring the 'decompressionTolerance' and 'decompressionRatioLimit' httpOption attributes to achieve the desired decompression results."
to,
"When this exception happens on a valid request body, reconfigure the 'decompressionTolerance' and 'decompressionRatioLimit' httpOption attributes to achieve the appropriate decompression results."

File: dev/com.ibm.ws.transport.http/resources/OSGI-INF/l10n/metatype.properties (hunk 1 above)
Comment (ManasiGandhi) [reply dev/com.ibm.ws.transport.http/resources/OSGI-INF/l10n/metatype.properties]: Made the changes above.

File: dev/com.ibm.ws.transport.http/resources/OSGI-INF/l10n/metatype.properties (hunk 2 above)
Comment (ManasiGandhi) [reply dev/com.ibm.ws.transport.http/resources/OSGI-INF/l10n/metatype.properties]: Updated the comment above.

**Comment by andrea-pichler** [comment]
Completed requested L2 message review

File: dev/com.ibm.ws.transport.http/resources/com/ibm/ws/http/channel/internal/resources/httpchannelmessages.nlsprops [hunk 6]
```diff
@@ -160,4 +160,8 @@ compression.duplicateOverwriteAndAdd.useraction=If the default configuration mus
 
 compression.unsupportedAlgorithm=CWWKT0033W: An attempt was made to configure {0} as the server-preferred algorithm. The supported values are gzip, x-gzip, zlib, deflate, identity, and none. The default configuration {1} is used instead.
 compression.unsupportedAlgorithm.explanation=The configuration is attempting to set an unsupported algorithm as the server-preferred algorithm to use for compression. 
-compression.unsupportedAlgorithm.useraction= When defining the value, ensure that it is set to gzip, x-gzip, deflate, zlib, identity, or none.
\ No newline at end of file
+compression.unsupportedAlgorithm.useraction= When defining the value, ensure that it is set to gzip, x-gzip, deflate, zlib, identity, or none.
```
[the remark this answers is not in the document]
Comment (mrsaldana) [reply dev/com.ibm.ws.transport.http/resources/com/ibm/ws/http/channel/internal/resources/httpchannelmessages.nlsprops]: This is not part of change being added, message is already out for this one.

File: dev/com.ibm.ws.transport.http/resources/com/ibm/ws/http/channel/internal/resources/httpchannelmessages.nlsprops (hunk 4 above)
Comment (mrsaldana) [reply dev/com.ibm.ws.transport.http/resources/com/ibm/ws/http/channel/internal/resources/httpchannelmessages.nlsprops]: Can we word this as, "The maximum number of decompression cycles resulting in a decompression ratio larger than the configured limit has been reached. The decompression limit is defined by the 'decompressionRatioLimit' httpOption attribute."

**Comment by mrsaldana** [comment]
#build

**Comment by mrsaldana** [comment]
#build

**Comment by mrsaldana** [comment]
#run-libby-bot

**Comment by mrsaldana** [comment]
fixes #10702

**Pull request merged**